baselines/diabetic_retinopathy_detection/radial.py

```diff
@@ -0,0 +1,356 @@
+  strategy = utils.init_distribution_strategy(FLAGS.force_use_cpu,
+                                              FLAGS.use_gpu, FLAGS.tpu)
+  use_tpu = not (FLAGS.force_use_cpu or FLAGS.use_gpu)
+
+  batch_size = FLAGS.batch_size * FLAGS.num_cores
+
+  # As per the Kaggle challenge, we have split sizes:
+  # train: 35,126
+  # validation: 10,906 (currently unused)
+  # test: 42,670
+  ds_info = tfds.builder('diabetic_retinopathy_detection').info
+  train_dataset_size = ds_info.splits['train'].num_examples
+  steps_per_epoch = train_dataset_size // batch_size
+  steps_per_eval = ds_info.splits['test'].num_examples // batch_size
+
+  dataset_train_builder = ub.datasets.get(
+      'diabetic_retinopathy_detection', split='train', data_dir=FLAGS.data_dir)
+  dataset_train = dataset_train_builder.load(batch_size=batch_size)
+  dataset_train = strategy.experimental_distribute_dataset(dataset_train)
+  dataset_test_builder = ub.datasets.get(
+      'diabetic_retinopathy_detection', split='test', data_dir=FLAGS.data_dir)
+  dataset_test = dataset_test_builder.load(batch_size=batch_size)
+  dataset_test = strategy.experimental_distribute_dataset(dataset_test)
+
+  if FLAGS.use_bfloat16:
+    policy = tf.keras.mixed_precision.experimental.Policy('mixed_bfloat16')
+    tf.keras.mixed_precision.experimental.set_policy(policy)
+
+  summary_writer = tf.summary.create_file_writer(
+      os.path.join(FLAGS.output_dir, 'summaries'))
+
+  with strategy.scope():
+    logging.info('Building Keras ResNet-50 Radial model.')
+    model = ub.models.resnet50_radial(
+        input_shape=utils.load_input_shape(dataset_train),
+        num_classes=1,  # binary classification task
+        prior_stddev=FLAGS.prior_stddev,
+        dataset_size=train_dataset_size,
+        stddev_init=FLAGS.stddev_init,
+        tied_mean_prior=FLAGS.tied_mean_prior,
+        radial_prior=FLAGS.radial_prior)
+
+    logging.info('Model input shape: %s', model.input_shape)
+    logging.info('Model output shape: %s', model.output_shape)
+    logging.info('Model number of weights: %s', model.count_params())
+
+    # Linearly scale learning rate and the decay epochs by vanilla settings.
+    base_lr = FLAGS.base_learning_rate * batch_size / DEFAULT_BATCH_SIZE
+    lr_decay_epochs = [
+        (int(start_epoch_str) * FLAGS.train_epochs) // DEFAULT_NUM_EPOCHS
+        for start_epoch_str in FLAGS.lr_decay_epochs]
+
+    lr_schedule = utils.LearningRateSchedule(
+        steps_per_epoch,
+        base_lr,
+        decay_ratio=FLAGS.lr_decay_ratio,
+        decay_epochs=lr_decay_epochs,
+        warmup_epochs=FLAGS.lr_warmup_epochs)
+    optimizer = tf.keras.optimizers.SGD(
+      lr_schedule, momentum=0.9, nesterov=True)
+    metrics = utils.get_diabetic_retinopathy_base_metrics(
+      use_tpu=use_tpu, num_bins=FLAGS.num_bins)
+    metrics.update({
+      'train/kl': tf.keras.metrics.Mean(),
+      'train/kl_scale': tf.keras.metrics.Mean()})
```

Comment:
Are these supposed to b ethe same?

Reply:
Yep, "Mean" just describes the aggregation applied when the metric is computed (we provide the KL and the KL scale respectively)